docs/generate-oval.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/python3
+# pylint: disable=invalid-name,missing-docstring
+#
+# Copyright (C) 2022 Richard Hughes <richard@hughsie.com>
+#
+# SPDX-License-Identifier: LGPL-2.1+
+
+import argparse
+import datetime
+import sys
+import json
+import os
+
+from typing import Dict, List, Any
+
+import xml.etree.ElementTree as ET
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-v", "--version", type=str, help="fwupd version")
+    parser.add_argument(
+        "filename_dst", action="store", type=str, help="XML destination"
+    )
+    parser.add_argument("json_attrs", nargs="+", help="JSON attributes")
+    args = parser.parse_args()
+
+    # parse JSON
+    items: List[str] = []
+    for fn in sorted(args.json_attrs):
+        try:
+            with open(fn, "rb") as f:
+                item = json.loads(f.read())
+        except json.decoder.JSONDecodeError as e:
+            print("failed to parse {}: {}".format(fn, str(e)))
+            sys.exit(1)
+        for tag in ["id", "name", "failure-results"]:
+            if tag not in item:
+                print("skipping {} as no {}".format(fn), tag)
+                continue
+        items.append(item)
+
+    oval_definitions = ET.Element("oval_definitions")
+    oval_definitions.set("xmlns", "http://oval.mitre.org/XMLSchema/oval-definitions-5")
+    oval_definitions.set("xmlns:oval", "http://oval.mitre.org/XMLSchema/oval-common-5")
+    oval_definitions.set(
+        "xmlns:unix-def", "http://oval.mitre.org/XMLSchema/oval-definitions-5#unix"
+    )
+    oval_definitions.set(
+        "xmlns:red-def", "http://oval.mitre.org/XMLSchema/oval-definitions-5#linux"
+    )
+    oval_definitions.set(
+        "xmlns:ind-def",
+        "http://oval.mitre.org/XMLSchema/oval-definitions-5#independent",
+    )
+    oval_definitions.set("xmlns:xsi", "http://www.w3.org/2001/XMLSchema-instance")
+    oval_definitions.set(
+        "xsi:schemaLocation",
+        "http://oval.mitre.org/XMLSchema/oval-common-5 oval-common-schema.xsd "
+        "http://oval.mitre.org/XMLSchema/oval-definitions-5 oval-definitions-schema.xsd "
+        "http://oval.mitre.org/XMLSchema/oval-definitions-5#unix unix-definitions-schema.xsd "
+        "http://oval.mitre.org/XMLSchema/oval-definitions-5#linux linux-definitions-schema.xsd",
+    )
+
+    generator = ET.SubElement(oval_definitions, "generator")
+    for key, value in {
+        "oval:product_name": "fwupd",
+        "oval:product_version": args.version,
+        "oval:schema_version": args.version,
```

Comment:
"schema_version" should address to OVAL schema version. For now, the HSI feature is implemented in version 5.11.3. We need a new argument to set OVAL schema version.

I only fixed this and then the generated XML file pass the "oscap oval validate"

Reply:
Fixed, thanks!